### PR TITLE
Feat/dtcore 2461

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | feat/DTCORE-2447
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | DTCORE-2461
| License          | BSD 3-Clause

## Description

- add region-selector in manager-pci-common
- add kube region selector to pci-kubernetes app
- refactor pci-block-storage app with the region-selector

### CODE REVIEW

**Review by kqesar — APPROVED** [review]

File: packages/manager/modules/manager-pci-common/src/components/region-selector/useRegionSelector.tsx [hunk 1]
```diff
@@ -16,10 +16,14 @@ export function useRegionSelector({
...
   const [selectedMicroRegion, setSelectedMicroRegion] = useState(null);
   const query = useProjectLocalisation(projectId);
-  const { continents, regions: unfilteredRegions } = query.data || {};
+  const { continents: unfilteredContinents, regions: unfilteredRegions } =
+    query.data || {};
   const regions = unfilteredRegions?.filter((r) =>
     regionFilter ? regionFilter(r) : true,
   );
+  const continents = unfilteredContinents?.filter(
+    (c) => c.id === 'WORLD' || regions.some((r) => r.continentLabel === c.name),
```
Comment (anooparveti) [inline packages/manager/modules/manager-pci-common/src/components/region-selector/useRegionSelector.tsx:25]: **nitpick:** Avoid shortnames.
```suggestion
    (continent) => continent.id === 'WORLD' || regions.some((region) => region.continentLabel === continent.name),
```

OR

```suggestion
    ({ id, name }) => id === 'WORLD' || regions.some(({ continentLabel }) => continentLabel === name),
```

File: packages/manager/modules/manager-pci-common/src/components/region-selector/RegionTile.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        color={ODS_THEME_COLOR_INTENT.text}
+        level={ODS_TEXT_LEVEL.body}
+        size={isSelected ? ODS_TEXT_SIZE._500 : ODS_TEXT_SIZE._400}
+      >
+        {region.isMacro ? region.macroLabel : region.microLabel}
+      </OsdsText>
+    </div>
+    {!isCompact && (
+      <>
+        <hr className="w-full border-solid border-0 border-b border-b-[#85d9fd]" />
```
Comment (anooparveti) [inline packages/manager/modules/manager-pci-common/src/components/region-selector/RegionTile.tsx:31]: **nitpick:** Can you replace it with suitable color from the ODS utilities?

File: packages/manager/modules/manager-pci-common/src/components/region-selector/useRegionSelector.tsx (hunk 1 above)
Comment (frenautvh) [reply packages/manager/modules/manager-pci-common/src/components/region-selector/useRegionSelector.tsx]: done

File: packages/manager/modules/manager-pci-common/src/components/region-selector/RegionTile.tsx (hunk 2 above)
Comment (frenautvh) [reply packages/manager/modules/manager-pci-common/src/components/region-selector/RegionTile.tsx]: done

**Review by seven-amid — APPROVED** [review]

**Comment by frenautvh** [comment]
rebased on 5190c56c0e625b67ca3bf35a2cbd4cffdbe7d7a6

**Review by anooparveti — APPROVED** [review]

**Review by anooparveti — CHANGES_REQUESTED** [review]
> Run Jest tests / test (push) Failing after 3m

Can you check this?

**Pull request merged**